src/lib/AST/ASTVisitor.cpp

```diff
@@ -521,6 +521,84 @@ class ASTVisitor
     {
         MRDOCS_ASSERT(D);
         MRDOCS_ASSERT(usr_.empty());
+
+        if (const auto* NAD = dyn_cast<NamespaceAliasDecl>(D))
+        {
+            if (index::generateUSRForDecl(cast<Decl>(NAD->getNamespace()), usr_))
+                return true;
+            usr_.append("@NA");
+            usr_.append(NAD->getNameAsString());
+            return false;
+        }
+
+        // Handling UsingDirectiveDecl
+        if (const auto* UDD = dyn_cast<UsingDirectiveDecl>(D))
+        {
+            if (index::generateUSRForDecl(UDD->getNominatedNamespace(), usr_)) {
+                return true;
+            }
+            usr_.append("@UD");
+            usr_.append(UDD->getQualifiedNameAsString());
+            return false;
+        }
+
+        // Handling UsingDecl
+        if (const auto* UD = dyn_cast<UsingDecl>(D))
+        {
+            for (const auto* shadow : UD->shadows())
+            {
+                if (index::generateUSRForDecl(shadow->getTargetDecl(), usr_))
+                    return true;
+            }
+            usr_.append("@UDec");
+            usr_.append(UD->getQualifiedNameAsString());
+            return false;
+        }
+
+        // Handling UnresolvedUsingTypenameDecl
+        if (const auto* UD = dyn_cast<UnresolvedUsingTypenameDecl>(D))
+        {
+            if (index::generateUSRForDecl(UD, usr_))
+                return true;
+            usr_.append("@UUTDec");
+            usr_.append(UD->getQualifiedNameAsString());
+            return false;
+        }
+
+        // Handling UnresolvedUsingValueDecl
+        if (const auto* UD = dyn_cast<UnresolvedUsingValueDecl>(D))
+        {
+            if (index::generateUSRForDecl(UD, usr_))
+                return true;
+            usr_.append("@UUV");
+            usr_.append(UD->getQualifiedNameAsString());
+            return false;
+        }
+
+        // Handling UsingPackDecl
+        if (const auto* UD = dyn_cast<UsingPackDecl>(D))
+        {
+            if (index::generateUSRForDecl(UD, usr_))
+                return true;
+            usr_.append("@UPD");
+            usr_.append(UD->getQualifiedNameAsString());
+            return false;
+        }
+
+        // Handling UsingEnumDecl
+        if (const auto* UD = dyn_cast<UsingEnumDecl>(D))
+        {
+            if (index::generateUSRForDecl(UD, usr_))
+                return true;
+            usr_.append("@UED");
+            EnumDecl const* ED = UD->getEnumDecl();
+            if (ED)
```

Comment:
Remove this branch, `ED` will never be `nullptr` (an assert will fail in clang if it is)